.travis.yml

```diff
@@ -16,7 +16,7 @@ matrix:
     - env: NAME='nightly'
       rust: nightly
     - env: NAME='rustfmt'
-      rust: nightly
+      rust: nightly-2018-05-14
```

Comment:
You want to keep just `nightly` here, and add `rust: nightly-2018-05-14` to the clippy build.

Reply:
But the `rustfmt` version is tied to a particular Rust nightly version, so if this is just `nightly` and in the `clippy` build the Rust version is fixed to `nightly-2018-05-14`, they diverge.

This has already happened: I used `nightly-2018-05-14` locally to make sure I can use the same Clippy version, and I used the corresponding `rustfmt`. That made the build fail because the _latest_ `rustfmt` already wanted to make some other changes!

It's really inconvenient if for every commit you make you have to switch between two Rust versions to make sure you apply the right `rustfmt` and `clippy`.

Reply:
Alternatively, I can upgrade the `clippy` build's Rust nightly version to a later one, and use a later Clippy, if you prefer.

Reply:
> It's really inconvenient if for every commit you make you have to switch between two Rust versions to make sure you apply the right rustfmt and clippy.

That is a good point! Maybe it's best then to pin the same nightly version for both rustfmt and clippy? Should release pinning also be done for rustfmt?

Another idea is just to keep both tools on their respective latest releases and building with nightly, and if clippy breaks, just temporarily add it to `allow_failures`.

What do you think @romac?

Reply:
> Should release pinning also be done for rustfmt?

I think that's automatic: Since (unlike clippy) rustfmt is installed via rustup, its version will always be the same for any given version of Rust itself. So with `nightly-2018-05-14`, we should always get rustfmt 0.7.0.

Reply:
I was looking for documentation on how rustup handled this actually earlier today. I didn't see it in the rustup README. Can you point me there? And how can I see exactly what version of a component will be built against a given Rust versiont? Thanks.